katib-ui-rock/rockcraft.yaml

```diff
@@ -19,36 +19,86 @@ platforms:
   amd64:
 
 parts:
-  katib-ui:
+  frontend-lib:
+    plugin: nil
+    source: https://github.com/kubeflow/kubeflow
+    source-type: git
+    # Set source-tag here to the commit cited in https://github.com/kubeflow/katib/blob/master/pkg/ui/v1beta1/frontend/COMMIT
+    # for this Katib release
+    # Note: Katib 0.16.0 cannot build with Kubeflow v1.8-branch because Kubeflow refactored this lib
+    source-commit: 046c6d3c8301a95a2809c0d927bbbf72638dd639
+    # Omit source-depth because we're cloning a commit
+    # source-depth: 1
+    build-environment:
+      # suppress analytics collection
+      - NG_CLI_ANALYTICS: false
+    build-snaps:
+      - node/12/stable
+    override-build: |
+      # Clean the build dir so we don't accidentally use files we didn't intend
+      rm -rf ./*
```

Comment:
Yeah this is more a me-pattern and I can remove it if needed.

By default, the `source: https://somewhere/in/github` directive pulls the source to `/root/parts/PARTNAME/src` and then copies that source to `/root/parts/PARTNAME/build`.  Most of our rocks then just `cd` to the dir that has the source and build there.

Upstream dockerfiles do this differently.  They don't pull the whole repo, but instead just `COPY` the files needed for the build (kinda like what I'm doing in this rockfile).  This felt more natural imo because it means I only build with the things that actually matter.

I'm paranoid (mainly because I don't know this software, and don't know if leaving extra files around could affect the build) so I went with the "copy only what I need" approach like the upstream dockerfile had.  That's why I do a `rm -rf ./`.  But it is probably overkill.  

wdyt?

Reply:
Yes, I think it's nice that we do it the way upstream does it. So IIUC `rm -rf ./` deletes the source code from `/root/parts/PARTNAME/build` and then copies only what we need right? If that's the case, I think we can keep it. To your concerns, having the repo's parent directories around wouldn't play any role since `npm build` uses specific files.

Reply:
Yes you're right about what is happening (remove source from `.../build`, then copy the files needed)